pkgs/by-name/bo/bochs/package.nix

```diff
@@ -3,20 +3,20 @@
 , fetchurl
 , SDL2
 , curl
+, darwin
 , docbook_xml_dtd_45
 , docbook_xsl
 , gtk3
 , libGL
 , libGLU
 , libX11
 , libXpm
-, libobjc
 , libtool
 , ncurses
 , pkg-config
 , readline
 , wget
-, wxGTK
+, wxGTK32
```

Comment:
Oh this is actually a bit concerning. This is a breaking change because `.override { wxGTX = ...; }` doesn't work anymore. I don't think we should make these changes until we have a better story for that, even if this allows removing definitions from `all-packages.nix`.

Note that even if you need custom arguments, you can still use the `pkgs/by-name` directory.

Reply:
Bochs didn't update their dependencies yet (they are more concerned about optimizing the emulation core), so overriding wxGTK to other version than that specific one will not work.

Reply:
Overrides can also be used to just make minor changes to packages, e.g.

```nix
.override (old: {
  wxGTK = old.wxGTK.overrideAttrs ...;
})
```

Similarly the `darwin` change could also cause breakage, consider:
```nix
.override {
  libobjc = ...;
}
```

These overrides would now silently not do anything anymore.

Of course, Nixpkgs has always not really enforced not having such breakages, but the migration to `pkgs/by-name` should not cause more of these to happen. This is a problem to be fixed in the future.

I opened https://github.com/NixOS/nixpkgs/pull/254632 to have this written down.

Reply:
> Similarly the darwin change could also cause breakage

That Darwin-specific things were always problematic to me, since the splicing incident (that, among other things, break `inherit  (X) a b c;` design pattern).

Reply:
Oh yeah that also messes with it, there's definitely many things to be improved. I feel like the custom argument one might be the next one to dig into.